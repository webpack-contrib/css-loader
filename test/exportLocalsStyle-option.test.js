import { webpack, evaluated } from './helpers';

describe('exportLocalsStyle option', () => {
  it('not specified', async () => {
    const config = { loader: { options: { modules: true } } };
    const testId = './modules/exportLocalsStyle.css';
    const stats = await webpack(testId, config);
    const { modules } = stats.toJson();
    const module = modules.find((m) => m.id === testId);
    const evaluatedModule = evaluated(module.source);

    expect(evaluatedModule).toMatchSnapshot('module (evaluated)');
    expect(evaluatedModule.locals).toMatchSnapshot('locals');
    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });

  it('asIs', async () => {
    const config = {
      loader: {
        options: {
          modules: true,
          exportLocalsStyle: 'asIs',
        },
      },
    };
    const testId = './modules/exportLocalsStyle.css';
    const stats = await webpack(testId, config);
    const { modules } = stats.toJson();
    const module = modules.find((m) => m.id === testId);
    const evaluatedModule = evaluated(module.source);

    expect(evaluatedModule).toMatchSnapshot('module (evaluated)');
    expect(evaluatedModule.locals).toMatchSnapshot('locals');
    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });

  it('camelCase', async () => {
    const config = {
      loader: {
        options: {
          modules: true,
          exportLocalsStyle: 'camelCase',
        },
      },
    };
    const testId = './modules/exportLocalsStyle.css';
    const stats = await webpack(testId, config);
    const { modules } = stats.toJson();
    const module = modules.find((m) => m.id === testId);
    const evaluatedModule = evaluated(module.source);

    expect(evaluatedModule).toMatchSnapshot('module (evaluated)');
    expect(evaluatedModule.locals).toMatchSnapshot('locals');
    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });

  it('camelCaseOnly', async () => {
    const config = {
      loader: {
        options: {
          modules: true,
          exportLocalsStyle: 'camelCaseOnly',
        },
      },
    };
    const testId = './modules/exportLocalsStyle.css';
    const stats = await webpack(testId, config);
    const { modules } = stats.toJson();
    const module = modules.find((m) => m.id === testId);
    const evaluatedModule = evaluated(module.source);

    expect(evaluatedModule).toMatchSnapshot('module (evaluated)');
    expect(evaluatedModule.locals).toMatchSnapshot('locals');
    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });

  it('dashes', async () => {
    const config = {
      loader: {
        options: {
          modules: true,
          exportLocalsStyle: 'dashes',
        },
      },
    };
    const testId = './modules/exportLocalsStyle.css';
    const stats = await webpack(testId, config);
    const { modules } = stats.toJson();
    const module = modules.find((m) => m.id === testId);
    const evaluatedModule = evaluated(module.source);

    expect(evaluatedModule).toMatchSnapshot('module (evaluated)');
    expect(evaluatedModule.locals).toMatchSnapshot('locals');
    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });

  it('dashesOnly', async () => {
    const config = {
      loader: {
        options: {
          modules: true,
          exportLocalsStyle: 'dashesOnly',
        },
      },
    };
    const testId = './modules/exportLocalsStyle.css';
    const stats = await webpack(testId, config);
    const { modules } = stats.toJson();
    const module = modules.find((m) => m.id === testId);
    const evaluatedModule = evaluated(module.source);

    expect(evaluatedModule).toMatchSnapshot('module (evaluated)');
    expect(evaluatedModule.locals).toMatchSnapshot('locals');
    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });
});
