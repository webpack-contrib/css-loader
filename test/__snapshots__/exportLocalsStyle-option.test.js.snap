// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exportLocalsStyle option asIs: errors 1`] = `Array []`;

exports[`exportLocalsStyle option asIs: locals 1`] = `
Object {
  "btn--info_is-disabled_1": "_1cmsy2E5swY7Ysn8h7qDun",
  "btn-info_is-disabled": "qysDcl74sBAzry5FpDPQi",
  "foo": "bar",
  "my-btn-info_is-disabled": "value",
  "simple": "f42SiWmkrcnocxM5cvNJw",
}
`;

exports[`exportLocalsStyle option asIs: module (evaluated) 1`] = `
Array [
  Array [
    1,
    ".qysDcl74sBAzry5FpDPQi {
  color: blue;
}

._1cmsy2E5swY7Ysn8h7qDun {
  color: blue;
}

.f42SiWmkrcnocxM5cvNJw {
  color: red;
}

a {
  color: yellow;
}
",
    "",
  ],
]
`;

exports[`exportLocalsStyle option asIs: warnings 1`] = `Array []`;

exports[`exportLocalsStyle option camelCase: errors 1`] = `Array []`;

exports[`exportLocalsStyle option camelCase: locals 1`] = `
Object {
  "btn--info_is-disabled_1": "_1cmsy2E5swY7Ysn8h7qDun",
  "btn-info_is-disabled": "qysDcl74sBAzry5FpDPQi",
  "btnInfoIsDisabled": "qysDcl74sBAzry5FpDPQi",
  "btnInfoIsDisabled1": "_1cmsy2E5swY7Ysn8h7qDun",
  "foo": "bar",
  "my-btn-info_is-disabled": "value",
  "myBtnInfoIsDisabled": "value",
  "simple": "f42SiWmkrcnocxM5cvNJw",
}
`;

exports[`exportLocalsStyle option camelCase: module (evaluated) 1`] = `
Array [
  Array [
    1,
    ".qysDcl74sBAzry5FpDPQi {
  color: blue;
}

._1cmsy2E5swY7Ysn8h7qDun {
  color: blue;
}

.f42SiWmkrcnocxM5cvNJw {
  color: red;
}

a {
  color: yellow;
}
",
    "",
  ],
]
`;

exports[`exportLocalsStyle option camelCase: warnings 1`] = `Array []`;

exports[`exportLocalsStyle option camelCaseOnly: errors 1`] = `Array []`;

exports[`exportLocalsStyle option camelCaseOnly: locals 1`] = `
Object {
  "btnInfoIsDisabled": "qysDcl74sBAzry5FpDPQi",
  "btnInfoIsDisabled1": "_1cmsy2E5swY7Ysn8h7qDun",
  "foo": "bar",
  "myBtnInfoIsDisabled": "value",
  "simple": "f42SiWmkrcnocxM5cvNJw",
}
`;

exports[`exportLocalsStyle option camelCaseOnly: module (evaluated) 1`] = `
Array [
  Array [
    1,
    ".qysDcl74sBAzry5FpDPQi {
  color: blue;
}

._1cmsy2E5swY7Ysn8h7qDun {
  color: blue;
}

.f42SiWmkrcnocxM5cvNJw {
  color: red;
}

a {
  color: yellow;
}
",
    "",
  ],
]
`;

exports[`exportLocalsStyle option camelCaseOnly: warnings 1`] = `Array []`;

exports[`exportLocalsStyle option dashes: errors 1`] = `Array []`;

exports[`exportLocalsStyle option dashes: locals 1`] = `
Object {
  "btn--info_is-disabled_1": "_1cmsy2E5swY7Ysn8h7qDun",
  "btn-info_is-disabled": "qysDcl74sBAzry5FpDPQi",
  "btnInfo_isDisabled": "qysDcl74sBAzry5FpDPQi",
  "btnInfo_isDisabled_1": "_1cmsy2E5swY7Ysn8h7qDun",
  "foo": "bar",
  "my-btn-info_is-disabled": "value",
  "myBtnInfo_isDisabled": "value",
  "simple": "f42SiWmkrcnocxM5cvNJw",
}
`;

exports[`exportLocalsStyle option dashes: module (evaluated) 1`] = `
Array [
  Array [
    1,
    ".qysDcl74sBAzry5FpDPQi {
  color: blue;
}

._1cmsy2E5swY7Ysn8h7qDun {
  color: blue;
}

.f42SiWmkrcnocxM5cvNJw {
  color: red;
}

a {
  color: yellow;
}
",
    "",
  ],
]
`;

exports[`exportLocalsStyle option dashes: warnings 1`] = `Array []`;

exports[`exportLocalsStyle option dashesOnly: errors 1`] = `Array []`;

exports[`exportLocalsStyle option dashesOnly: locals 1`] = `
Object {
  "btnInfo_isDisabled": "qysDcl74sBAzry5FpDPQi",
  "btnInfo_isDisabled_1": "_1cmsy2E5swY7Ysn8h7qDun",
  "foo": "bar",
  "myBtnInfo_isDisabled": "value",
  "simple": "f42SiWmkrcnocxM5cvNJw",
}
`;

exports[`exportLocalsStyle option dashesOnly: module (evaluated) 1`] = `
Array [
  Array [
    1,
    ".qysDcl74sBAzry5FpDPQi {
  color: blue;
}

._1cmsy2E5swY7Ysn8h7qDun {
  color: blue;
}

.f42SiWmkrcnocxM5cvNJw {
  color: red;
}

a {
  color: yellow;
}
",
    "",
  ],
]
`;

exports[`exportLocalsStyle option dashesOnly: warnings 1`] = `Array []`;

exports[`exportLocalsStyle option not specified: errors 1`] = `Array []`;

exports[`exportLocalsStyle option not specified: locals 1`] = `
Object {
  "btn--info_is-disabled_1": "_1cmsy2E5swY7Ysn8h7qDun",
  "btn-info_is-disabled": "qysDcl74sBAzry5FpDPQi",
  "foo": "bar",
  "my-btn-info_is-disabled": "value",
  "simple": "f42SiWmkrcnocxM5cvNJw",
}
`;

exports[`exportLocalsStyle option not specified: module (evaluated) 1`] = `
Array [
  Array [
    1,
    ".qysDcl74sBAzry5FpDPQi {
  color: blue;
}

._1cmsy2E5swY7Ysn8h7qDun {
  color: blue;
}

.f42SiWmkrcnocxM5cvNJw {
  color: red;
}

a {
  color: yellow;
}
",
    "",
  ],
]
`;

exports[`exportLocalsStyle option not specified: warnings 1`] = `Array []`;
